webapp/src/errors.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useIntl} from 'react-intl'
+
+enum ErrorId {
+    WorkspaceUndefined = 'workspace-undefined',
+    NotLoggedIn = 'not-logged-in',
+}
+
+type ErrorDef = {
+    title: string
+
+    button1Enabled: boolean
+    button1Text: string
+    button1Redirect: string | (() => string)
+    button1Fill: boolean
+
+    button2Enabled: boolean
+    button2Text: string
+    button2Redirect: string | (() => string)
+    button2Fill: boolean
+}
+
+function errorDefFromId(id: ErrorId | null): ErrorDef {
+    const errDef = {
+        title: '',
+        button1Enabled: false,
+        button1Text: '',
+        button1Redirect: '',
+        button1Fill: false,
+        button2Enabled: false,
+        button2Text: '',
+        button2Redirect: '',
+        button2Fill: false,
+    }
+
+    const intl = useIntl()
+
+    switch (id) {
+    case ErrorId.WorkspaceUndefined: {
+        errDef.title = intl.formatMessage({id: 'error.workspace-undefined', defaultMessage: 'Not a valid workspace.'})
```

Comment:
Is it possible to make this friendlier? For example, can we say "The workspace you're looking for may have been deleted or is not valid." 
Are there any remediation steps for it or is "Go to Dashboard" the next step?

Reply:
I don't think we can try and guess which workspace they meant, or even send them to one that exists as that would leak the id.